### PR TITLE
Add release notes for ML.NET 0.6

### PR DESCRIPTION
**Pull request by GalOshri**
This adds release notes for ML.NET 0.6.

### CODE REVIEW

**Comment by codemzs** [comment]
For type system you can add issue #673 that contains the spec and benchmarks.

File: docs/release-notes/0.6/release-0.6.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+```
+
+### Release Notes
+
+Below are some of the highlights from this release.
+
+* New APIs for ML.NET
+    
+    * While the `LearningPipeline` APIs that were released with ML.NET 0.1 were easy to get started with, they had obvious limitations in functionality. Certain tasks that were possible with the internal version of ML.NET like inspecting model weights, creating a transform-only pipeline, and training from an initial predictor could not be done with `LearningPipeline`.
+    * The important concepts for understanding the new API are introduced [here](https://github.com/dotnet/machinelearning/blob/3cdd3c8b32705e91dcf46c429ee34196163af6da/docs/code/MlNetHighLevelConcepts.md).
```
Comment (Zruty0) [inline docs/release-notes/0.6/release-0.6.md]: >3cdd3c8b32705e91dcf46c429ee34196163af6da [](start = 128, length = 40)

now that it's merged, you should use [https://github.com/dotnet/machinelearning/blob/master/docs/code/MlNetHighLevelConcepts.md](https://github.com/dotnet/machinelearning/blob/master/docs/code/MlNetHighLevelConcepts.md) etc.

File: docs/release-notes/0.6/release-0.6.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    * The [TensorFlow scoring transform](https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.transforms.tensorflowtransform?view=ml-dotnet) released in ML.NET 0.5 enabled using 'frozen' TensorFlow models. In ML.NET 0.6, 'saved' TensorFlow models can also be used. 
+    * An API was added to extract information about the nodes in a TensorFlow model. This can help identifying the input and output of a TensorFlow model. Example usage can be found [here](https://github.com/dotnet/machinelearning/blob/3cdd3c8b32705e91dcf46c429ee34196163af6da/src/Microsoft.ML.DnnAnalyzer/Microsoft.ML.DnnAnalyzer/DnnAnalyzer.cs).
+
+* Replaced ML.NET's Dv type system with .NET's standard type system ([#863](https://github.com/dotnet/machinelearning/pull/863))
+
+    * ML.NET previously had its own type system which helped it more efficiently deal with things like missing values (a common case in ML). This type system required users to work with types like `DvText`, `DvBool`, `DvInt4`, etc. 
+    * This update replaces the Dv type system with .NET's standard type system to make ML.NET easier to use and to take advantage of innovation in .NET.
```
Comment (Zruty0) [inline docs/release-notes/0.6/release-0.6.md]: >easier to use [](start = 94, length = 13)

probably mention the fact that now only floats and doubles have missing values, represented with NaN?

File: docs/release-notes/0.6/release-0.6.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+* Up to ~200x speedup in prediction engine performance ([#973](https://github.com/dotnet/machinelearning/pull/973))
+
+    * This improvement leads to a significant speedup when making predictions for single records.
```
Comment (Zruty0) [inline docs/release-notes/0.6/release-0.6.md]: >This improvement leads to a significant speedup [](start = 6, length = 47)

LOL. First rule of tautology club is the first rule of tautology club.

File: docs/release-notes/0.6/release-0.6.md (hunk 2 above)
Comment (GalOshri) [reply docs/release-notes/0.6/release-0.6.md]: Addressed.

File: docs/release-notes/0.6/release-0.6.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+Below are some of the highlights from this release.
+
+* New APIs for ML.NET
+    
+    * While the `LearningPipeline` APIs that were released with ML.NET 0.1 were easy to get started with, they had obvious limitations in functionality. Certain tasks that were possible with the internal version of ML.NET like inspecting model weights, creating a transform-only pipeline, and training from an initial predictor could not be done with `LearningPipeline`.
+    * The important concepts for understanding the new API are introduced [here](https://github.com/dotnet/machinelearning/blob/3cdd3c8b32705e91dcf46c429ee34196163af6da/docs/code/MlNetHighLevelConcepts.md). 
+    * A cookbook that shows how to use these APIs for a variety of existing and new scenarios can be found [here](https://github.com/dotnet/machinelearning/blob/3cdd3c8b32705e91dcf46c429ee34196163af6da/docs/code/MlNetCookBook.md). 
+    * These APIs are still evolving, so we would love to hear any feedback or questions. 
+    * The `LearningPipeline` APIs have moved to the `Microsoft.ML.Legacy` namespace.
```
Comment (eerhardt) [inline docs/release-notes/0.6/release-0.6.md]: I would mention that we plan on removing the `LearningPipeline` APIs in the next `0.7` release.

File: docs/release-notes/0.6/release-0.6.md (hunk 2 above)
Comment (chris-lauren) [reply docs/release-notes/0.6/release-0.6.md]: @GalOshri can you also add pointer to sample for ONNX scoring and if we have any data re: the perf speedup for prediction can we share? I feel uncomfortable explaining a big perf speedup with no backing evidence :(

**Pull request merged**